controlplane/test/graph-pruning.test.ts

```diff
@@ -0,0 +1,541 @@
+    const { client, authenticator, server } = await SetupTest({
+      dbname,
+      setupBilling: { plan: 'launch@1' },
+      enableMultiUsers: true,
+    });
+    const response = await client.enableGraphPruning({
+      enableGraphPruning: true,
+      namespace: 'default',
+    });
+
+    expect(response.response?.code).toBe(EnumStatusCode.OK);
+
+    authenticator.changeUser(TestUser.viewerTimCompanyA);
+
+    const graphPruningConfigs = [
+      {
+        ruleName: 'UNUSED_FIELDS',
+        severityLevel: LintSeverity.error,
+        gracePeriodInDays: 7,
+      } as GraphPruningConfig,
+      {
+        ruleName: 'DEPRECATED_FIELDS',
+        severityLevel: LintSeverity.warn,
+        gracePeriodInDays: 7,
+      } as GraphPruningConfig,
+      {
+        ruleName: 'REQUIRE_DEPRECATION_BEFORE_DELETION',
+        severityLevel: LintSeverity.warn,
+        gracePeriodInDays: 7,
+      } as GraphPruningConfig,
+    ];
+
+    const configureGraphPruningConfigResponse = await client.configureNamespaceGraphPruningConfig({
+      namespace: 'default',
+      configs: graphPruningConfigs,
+    });
+
+    expect(configureGraphPruningConfigResponse.response?.code).toBe(EnumStatusCode.ERR);
+    expect(configureGraphPruningConfigResponse.response?.details).toBe(
+      'The user doesnt have the permissions to perform this operation',
+    );
+
+    await server.close();
+  });
+
+  test('Should configure graph pruning config, run check subgraph command and verify the results', async (testContext) => {
+    const { client, server } = await SetupTest({ dbname, setupBilling: { plan: 'launch@1' }, chClient });
+
+    const federatedGraphName = genID('fedGraph');
+    const subgraphName = genID('subgraph1');
+    const label = genUniqueLabel();
+
+    const createFederatedGraphResp = await client.createFederatedGraph({
+      name: federatedGraphName,
+      namespace: 'default',
+      labelMatchers: [joinLabel(label)],
+      routingUrl: 'http://localhost:8081',
+    });
+    expect(createFederatedGraphResp.response?.code).toBe(EnumStatusCode.OK);
+
+    let resp = await client.createFederatedSubgraph({
+      name: subgraphName,
+      namespace: 'default',
+      labels: [label],
+      routingUrl: 'http://localhost:8080',
+    });
+
+    expect(resp.response?.code).toBe(EnumStatusCode.OK);
+
+    resp = await client.publishFederatedSubgraph({
+      name: subgraphName,
+      namespace: 'default',
+      schema: initSchema,
+    });
+
+    expect(resp.response?.code).toBe(EnumStatusCode.OK);
+
+    const response = await client.enableGraphPruning({
+      enableGraphPruning: true,
+      namespace: 'default',
+    });
+    expect(response.response?.code).toBe(EnumStatusCode.OK);
+
+    const graphPruningConfigs = [
+      {
+        ruleName: 'UNUSED_FIELDS',
+        severityLevel: LintSeverity.error,
+        gracePeriodInDays: 7,
+      } as GraphPruningConfig,
```

Comment:
Please try to avoid casting types because it is not type safe.

Reply:
Sorry, i remember to not do it, but as its a proto property. Doing it like
`const a:GraphPruningConfig={}` will give an error as we wont pass the proto attributes.
But have a soln `new GraphPruningConfig({})`

Done